### PR TITLE
Dotnet5 wizard

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Create our own LambdaRuntime. In the future this can be expanded as needed
- Plumb LambdaRuntime through the project wizard and as little of the rest of the code as possible
- Split supportedRuntimes/supportedPackagingTypes into supportedZipRuntimes and supportedImageRuntimes

Out of scope:
- On top of this PR we need a way to gate creation to different sam versions (leveraging LambdaRuntime)
- We should move all of our code to use our own LambdaRuntime instead of the sdk Runtime
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfigurationProducer.kt [hunk 1]
```diff
@@ -60,7 +61,7 @@ class LocalLambdaRunConfigurationProducer : LazyRunConfigurationProducer<LocalLa
         val runtime = RuntimeGroup.determineRuntime(context.module) ?: RuntimeGroup.determineRuntime(context.project)
 
         setAccountOptions(configuration)
-        configuration.useHandler(runtime, handler)
+        configuration.useHandler(Runtime.fromValue(runtime.toString()), handler)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfigurationProducer.kt:64]: Should this be .toSdkRuntime()?

**Review by abrooksv — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfigurationProducer.kt (hunk 1 above)
Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfigurationProducer.kt:64]: Ah nice catch, it should just be `runtime`. I converted this path to LambdaRuntime but we don't need to yet

File: core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    JAVA11(Runtime.JAVA11),
+    PYTHON2_7(Runtime.PYTHON2_7),
+    PYTHON3_6(Runtime.PYTHON3_6),
+    PYTHON3_7(Runtime.PYTHON3_7),
+    PYTHON3_8(Runtime.PYTHON3_8),
+    DOTNETCORE2_1(Runtime.DOTNETCORE2_1),
+    DOTNETCORE3_1(Runtime.DOTNETCORE3_1),
+    DOTNET5_0(null, "dotnet5.0");
+
+    override fun toString() = runtime?.toString() ?: runtimeOverride ?: throw IllegalStateException("LambdaRuntime has no runtime or override string")
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt:22]: Sum types would be useful here, but yeah it's a bit weird. The public interface is currently just `toString()` `toSdkRuntime()` and `fromValue` though so we can mess with the internals later if they prove too funky

File: jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt [hunk 3]
```diff
@@ -58,7 +56,8 @@ class DotNetSamProjectGenerator(
     private val generator = SamProjectGenerator()
     private val samPanel = SamInitSelectionPanel(generator.wizardFragments) {
         // Only show templates for DotNet in Rider
-        RuntimeGroup.getById(BuiltInRuntimeGroups.Dotnet).runtimes.contains(it)
+        // TODO fix this to work properly, the issue is that RuntimeGroup contains the sdk built in Runtime
```
[the remark this answers is not in the document]
Comment (hunterwerlla) [reply jetbrains-rider/src-201-202/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamProjectGenerator.kt:59]: Added a better explination

File: core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (abrooksv) [reply core/src/software/aws/toolkits/core/lambda/LambdaRuntime.kt:22]: I played with having 2 constructors on the enum (with a single backing field that was a String) but it made `toSdkRuntime()` a pain.

**Pull request merged**